addons/mrp/models/mrp_production.py

```diff
@@ -973,6 +972,10 @@ def _autoconfirm_production(self):
             moves_to_confirm._action_confirm()
             # run scheduler for moves forecasted to not have enough in stock
             moves_to_confirm._trigger_scheduler()
+            if production.picking_type_id.reservation_method == 'at_confirm'\
```

Comment:
You are outside the loop on `self` so `production` will be the last one of the set. Maybe you should group `moves_to_confirm` by picking_type_id
If I don't think we will ever see different manufacturing picking types, but we never know ...

Reply:
I'm a dummy. This should be fixed with the `stock_move.action_confirm` fixup.